### ISSUE TITLE
Fix and test spec contributing docs

### ISSUE THREAD

**Issue by bollwyvl** [krassowski/jupyterlab-lsp#317]
## What I am trying to do...

Write a spec for a new language server.

- comes from #313

## How I would like to learn how to do it...

Create a more extensive example of a full spec in the docs

## How the project might keep the docs accurate...

- generate a working example from the examples we test
- add a static fixture that get tested
- add a utility for humanely validating a/all specs without starting the server and debugging
  - could actually initialize the servers, dump out capabilities

**Comment by krassowski** [comment]
Please let's consider removing the docs page to "Add new LSP server" or something along the lines rather than the current "Configuration"

**Comment by bollwyvl** [comment]
Sure: maybe "Provisioning" or "Connecting" or something. We should try
_not_ to have aronyms in page titles/headers, if we can avoid it. On the
other hand, maybe it makes sense to rename the other thing "Settings".
Because it's called Settings. In the UI. On the wire, sure, it's
configurationChanged, etc.

However, the naming dates back to being a standalone package before we
merged the docs, which is in fact configured, with a file called
"jupyter_notebook_config" vs.... some thing you do in your browser on a
thing called settings... in some place... that puts something called
"settings"... on your disk... somewhere... i'm definitely _not_ advocating
for moving any of the server stuff there, any time soon, as it's really not
very reproducible, and, as I have mentioned, would actually prefer to be
able to _also_ put the settings schema on the traitlets config, with the
lab settings winning.

**Comment by krassowski** [comment]
Yes, using settings consistently is a good idea. "Connecting" has a similar problem of two meanings (connecting a server, connecting to a server), "Provisioning" is not great for non-native speakers (but might be good enough!). "Adding new language"? Also, maybe move it to "Advanced"?

**Comment by krassowski** [comment]
I think we should mention the requirement of having properly set mime-type in `languageInfo` (with a link to the official jupyter docs). I can later write a paragraph on the potential need of having post-processing of the virtual files to make the LSP work in notebooks.

**Comment by bollwyvl** [comment]
To the last point of the original post: more accurate schema (even if it's back-filling the truth), automation tools, logs, and hoisting said logs, are probably better than documentation-for-documentations-sake which, as we have seen, a lot of folk don't read anyway. We _could_ provide a link to the docs _in_ said log messages, but we're in fact talking about _renaming_ links here, so maybe that's not the best idea.
